docs/AQL/master03-syntax.adoc

```diff
@@ -475,6 +491,72 @@ The `AVG()` function returns the average value of given _expression_ argument. T
 Input values type should be either Integer of Real, and it will also determine the return type.
 
+=== Other functions
+
+==== TERMINOLOGY
+
+Often the `matches` operator may be need to perform a request to a terminology server, in order to execute some _operations_ over a value set or code system (a synonym of _terminology_).
+Several operations may be possible: the expansion of a ValueSet (or Reference Set), checking that a concept belongs to a value set or code system, testing if one concept subsumes another one, or the translation of concepts across different terminologies.
+While matching value by a pure terminology concept specified as <<_matches_terminology_uri,a terminology URI>> is possible, the invocation of external terminology servers from AQL requires a special function, to perform all these operations.
+
+The `TERMINOLOGY` function is used to invoke operations of an external terminology server and return the result.
+The syntax is `TERMINOLOGY(operation, driver, params_uri)`, having all arguments of type String, while the return type depends on the invoked `operation` and `driver` and is considered to be of `Any` type.
+
+The `operation` argument specifies the action to perform over the specified value set or code system. It is not restricted to any particular value as different terminology servers may use different ways of specifying the operation and its parameters. The specified operation can be properly dereferenced by the implementation of the `driver`.
+Some examples of mostly used operations are:
+
+* *expand*: Expand a value set; this will retrieve all the codes contained in a value set as an explicit set.
+* *validate*: Validate a code in a value set; this will check if a given code belongs to a value set. Recall that the value set may comprise all the codes in a code system (terminology).
+* *lookup*: Look-up a code; this will retrieve all the information concerning one particular code. Examples are retrieving the preferred form to display, synonyms, etc.
+* *translate*: Translate a code; this will translate (find an equivalent) code from one Value Set to another one based on a predefined mapping available in the external terminology service. Translation precision may not be limited to full equivalence and different kinds of mappings may be possible (wider meaning, equivalent, narrower meaning, etc).
+* *subsumes*: Subsumption testing; this will determine if a particular terminology concept is a subtype (is-a) of another one. For example, test in SNOMED-CT if `Myasthenia Gravis | 91637004` is a subtype of `autoimmune disease | 85828009` (i.e. test if 85828009 subsumes 91637004).
+
+The `driver` argument represents an identifier of the terminology server API. It provides the information to send requests related to the ValueSet and operation parameters to the the right terminology service. In addition, it allows the system to understand how to parse responses back from the terminology service.
```

Comment:
I don't remember why this is called 'driver', and it still doesn't make sense to me - is there no better name like 'service' or similar?

Reply:
that was the name stated in https://openehr.atlassian.net/wiki/spaces/spec/pages/624754994/Terminology+Server+Invocation+in+AQL+SEC+proposal

Reply:
I think it was Heath's suggestion and the idea was more to suggest an adaptor kinda like JDBC/ODBC, rather than specific service it is absolutely not intended as specific service endpoint URL - simply a namespace URI for a known type of 'terminology syntax processor'

Reply:
I thought that might be the case, but probably worth clarifying that point?

Reply:
I think we need to be 100% clear on what formal entities these URIs are identifying. APIs? Model of terminology (e.g. a UML model or schema)? Syntax?

Reply:
Ok, agree that it needs more clarification. I did changes, hopefully now is a bit clear.
About formal UML or syntax for these external entities: I guess it would be hard to make and maintain in time. I have the impression that the team (that made the original doc) wished this to be left open.

Reply:
agree with Thomas, this is the name kept just because it was the used before I was involved. I woul suggest something like  "Server flavor", "Server type", "Server API flavor" or similar.

Reply:
on, changed `driver` to `service_api`